diff --git a/crypto/x509v3/v3_purp.c b/crypto/x509v3/v3_purp.c
index f59bfc184..7e564dc35 100644
--- a/crypto/x509v3/v3_purp.c
+++ b/crypto/x509v3/v3_purp.c
@@ -292,6 +292,7 @@ int X509_supported_extension(X509_EXTENSION *ex)
 		NID_basic_constraints,	/* 87 */
 		NID_certificate_policies, /* 89 */
         	NID_ext_key_usage,	/* 126 */
+		NID_qcStatements, /* 286 */
 #ifndef OPENSSL_NO_RFC3779
 		NID_sbgp_ipAddrBlock,	/* 290 */
 		NID_sbgp_autonomousSysNum, /* 291 */
